### PR TITLE
Creating async options error messaging for Facebook Groups

### PR DESCRIPTION
**Pull request by malexanderlim**
### <samp>🤖 Generated by Copilot at d38e31b</samp>

Improved error handling for Facebook Groups source. The source now checks if the selected group has the Pipedream app installed and shows a helpful message if not.

### <samp>🤖 Generated by Copilot at d38e31b</samp>

> _Oh we are the coders of the Pipedream app_
> _And we work on the `postId` prop_
> _We handle the errors when the group is wrong_
> _And we tell the user how to get along_

## WHY

Screenshot 2023-06-21 at 12 16 43 PM

### <samp>🤖 Generated by Copilot at d38e31b</samp>

* Import and throw `ConfigurationError` for groups without Pipedream app ([link](https://github.com/PipedreamHQ/pipedream/pull/6972/files?diff=unified&w=0#diff-8e811b19be1470e1e31109e1a0191d72054a0c1e1466d194ee24e7be45bb58ddL1-R4), [link](https://github.com/PipedreamHQ/pipedream/pull/6972/files?diff=unified&w=0#diff-8e811b19be1470e1e31109e1a0191d72054a0c1e1466d194ee24e7be45bb58ddL30-R47))

### CODE REVIEW

File: components/facebook_groups/facebook_groups.app.mjs [hunk 1]
```diff
@@ -27,16 +30,21 @@ export default {
...
+          const { data } = await this.listPosts(opts);
+          return data?.map(({
+            id, message,
+          }) => ({
+            label: message || id,
+            value: id,
+          })) || [];
+        } catch (error) {
+          console.error(error);
+          throw new ConfigurationError("Please doublecheck that the Pipedream app has been installed in the group you have selected above. Learn more [here].(https://pipedream.com/apps/facebook-groups)");
```
Comment (dylburger) [inline components/facebook_groups/facebook_groups.app.mjs:46]:
```suggestion
throw new ConfigurationError("Please double-check that the Pipedream app has been installed in the group you have selected above. Learn more [here](https://pipedream.com/apps/facebook-groups).");
```

Comment (dylburger) [reply components/facebook_groups/facebook_groups.app.mjs:46]: Changed 'doublecheck' to 'double-check' according to Google's Developer docs style guide.

File: components/facebook_groups/facebook_groups.app.mjs [hunk 2]
```diff
@@ -27,16 +30,21 @@ export default {
...
+          const { data } = await this.listPosts(opts);
+          return data?.map(({
+            id, message,
+          }) => ({
+            label: message || id,
+            value: id,
+          })) || [];
+        } catch (error) {
+          console.error(error);
+          throw new ConfigurationError("Please doublecheck that the Pipedream app has been installed in the group you have selected above. Learn more [here](https://pipedream.com/apps/facebook-groups).");
```
Comment (dylburger) [inline components/facebook_groups/facebook_groups.app.mjs:46]:
```suggestion
          throw new ConfigurationError("Please double-check that the Pipedream app has been installed in the group you have selected above. Learn more [here](https://pipedream.com/apps/facebook-groups).");
```

Comment (dylburger) [reply components/facebook_groups/facebook_groups.app.mjs:46]: Changed 'doublecheck' to 'double-check' to follow Google's Developer docs style guide (hyphenation).

File: components/facebook_groups/facebook_groups.app.mjs [hunk 3]
```diff
@@ -27,16 +30,21 @@ export default {
...
+          const { data } = await this.listPosts(opts);
+          return data?.map(({
+            id, message,
+          }) => ({
+            label: message || id,
+            value: id,
+          })) || [];
+        } catch (error) {
+          console.error(error);
+          throw new ConfigurationError("Please double-check that the Pipedream app has been installed in the group you have selected above. Learn more [here](https://pipedream.com/apps/facebook-groups#getting-started).");
```
Comment (dylburger) [inline components/facebook_groups/facebook_groups.app.mjs:46]:
```suggestion
          throw new ConfigurationError("Please double-check that the Pipedream app has been installed in the group you have selected above. Learn more [here](https://pipedream.com/apps/facebook-groups#getting-started).");
```

Comment (dylburger) [reply components/facebook_groups/facebook_groups.app.mjs:46]: Changed 'double-check' to 'double-check' to follow Google's Developer docs style guide (https://developers.google.com/style/hyphens)

**Review by dannyroosevelt — COMMENTED** [review]
is that `post` prop common in all components? is that the only place you need to catch that?

**Comment by malexanderlim** [comment]
I've added error messaging for all actions.

**Comment by malexanderlim** [comment]
I've added error messaging for all actions.

File: components/facebook_groups/facebook_groups.app.mjs [hunk 4]
```diff
@@ -7,7 +10,7 @@ export default {
     group: {
       type: "string",
       label: "Group",
-      description: "Identifier of a group",
+      description: "Identifier of a group. The Pipedream app must be installed in the Facebook group in order to make any API calls or receive data.",
       async options() {
         const userId = await this.getUserId();
         const opts = {
```
Comment (michelle0927) [inline components/facebook_groups/facebook_groups.app.mjs:16]: You could add a try/catch block in the `group` async options, just like in the `post` async options. Both of the sources have a `group` prop, so if they received an error, the user would see the error message & link while setting up the props, and wouldn't be able to deploy.

**Review by michelle0927 — APPROVED** [review]
Looks good. I left a comment/suggestion about displaying the error msg & link for sources.

File: components/facebook_groups/facebook_groups.app.mjs (hunk 4 above)
Comment (malexanderlim) [reply components/facebook_groups/facebook_groups.app.mjs:16]: I think the issue with this is that the groups async option which uses `listGroups` wouldn't actually fail if the Pipedream app wasn't installed in the group, so it differs as compared to the `post` async option where `listPosts` actually requires the prior installation. 

I will merge this PR for now and come back to error message for sources, but I think that having the similar try/catch block on `groups` would not actually flag the lack of installation of the app given what the `groups` async options actually calls.

**Pull request merged**